File: test7C#/test7/Model/Donut/GingerDonut.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace test7.Model.Donut
{
    class GingerDonut : SimpleDonut
    {
        public GingerDonut() : base("Ron")
        {
            base.Flavor("Red hairs");
        }
    }
}

File: test7Go/oop/models/donut/GingerDonut.go
package donut

type GingerDonut struct {
	SimpleDonut
}

func NewGingerDonut() GingerDonut {
	simpleDonut := NewSimpleDonut("Ron")
	simpleDonut.Flavor = "Red hairs"

	return GingerDonut{simpleDonut}
}
